README.md

```diff
@@ -20,3 +20,32 @@ randomly without memory instead of systematically trying them?
 
 Describe your reasoning and the conclusion you've come to. Your reasoning is the
 most important part. Add your answer to this markdown file.
+
+
+# My Runtime Analysis
+
+In the base case, where the length of the array is 0 or 1, the runtime is constant,
+since in these scenarios, the array is already sorted.
+
+However, when the array size is greater than 1, the permutate function is called. This function generates every possible permutation of the array. With an array that a length of $n$, essentially the algorith will walk through array and swap elements to generate a unique permutation. 
+
+We start with the first permutation, so there is $n$ choices. For the second permutation, there is $n-1$ choice, and so on for the third ($n-2$ choices), fourth ($n-3$ choices), fifth ($n-4$ choices), etc... And since every permutation depends on the one before it, we can write this out as:
+
+$$n*(n-1)*(n-2)*(n-3)... = n!$$
+
+In the worst case scenario, after the permutate function finishes generating all permutations, all the permutations are checked to find the sorted permutation (for the wost case scenario, the sorted permutation is last one to be checked), taking a time complexity of $n$. In the isSolve() method, I have to convert from a string to an array which increases the time complexity to $2n$ for finding which permutation is sorted. Combining these terms, we get $\Theta(2n + n!)$ as worst case complexity for my algorithm. 
```

Comment:
Still don't see how you get n for checking all permutations.

Reply:
I'm going to reference an older submission, Submission 2 (which you can find above), because I believe that version is closer to what I currently believe...

Looking within the code.js file, at line 12, we see that we iterate through the allPermuations list $n$ times. The isSorted() method by itself only iterates through the array $n$ times as well.

The reason I state that the isSorted() method is $n^2$ is because when it is initially called at line 13, this call is nested within a $n$ length loop at line 12. Perhaps it would not be accurate to say the isSorted() itself is $n^2$, but $n^2$ is the complexity for the loop at line 12.

Is this line of thinking flawed?

Reply:
That sounds right; what I mean is the first sentence, which seems to say that generating all permutations and checking them takes n time. This clearly can't be, as there are more than n permutations.

Reply:
I suppose that it would also have to be $n!$, right? It was incorrect to state that the list of all permutation is of length $n$, but instead it would be a length of $n!$?

If this was the case, then the final complexity would resemble $\Theta(n * n! + n!)$ (or $\Theta(n* n!)$)

Reply:
That sounds like it's going in the right direction, but make sure that you're very clear when you're referring the number of permutations, the work done for each permutation, or the combination of the two.